### PR TITLE
feat(Algebra/Group/Units): Add a lemma `isUnit_map_iff'`

### PR DESCRIPTION
**Pull request by jjdishere**
Add a lemma `isUnit_map_iff'`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Group/Units/Hom.lean [hunk 1]
```diff
@@ -172,6 +172,11 @@ theorem _root_.isUnit_map_of_leftInverse [MonoidHomClass F M N] [MonoidHomClass
     {f : F} {x : M} (g : G) (hfg : Function.LeftInverse g f) :
     IsUnit (f x) ↔ IsUnit x := ⟨of_leftInverse g hfg, map _⟩
 
+@[to_additive]
+theorem _root_.map_isUnit_iff [MulEquivClass E M N]
```
Comment (eric-wieser) [inline Mathlib/Algebra/Group/Units/Hom.lean:176]: I think the name should be:
```suggestion
theorem _root_.isUnit_map_iff [MulEquivClass E M N]
```

Comment (jjdishere) [reply Mathlib/Algebra/Group/Units/Hom.lean:176]: Thank you!

**Comment by jjdishere** [comment]
`isUnit_map_iff` is already declared as 
```
@[simp]
theorem isUnit_map_iff (f : R →+* S) [IsLocalRingHom f] (a) : IsUnit (f a) ↔ IsUnit a
```

**Comment by eric-wieser** [comment]
Can you start a zulip thread suggesting that we generalize `IsLocalRingHom` to `IsLocalMonoidHom` or something? It should work fine on MonoidHom too...

**Comment by jjdishere** [comment]
This PR is obsolete due to #6045 . This theorem is achieved by stating `isUnit_map_iff` for every `IsLocalHom f`, and add the instance for any `MulEquiv` being `IsLocalHom`.

**Pull request closed without merging**